src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4

```diff
@@ -120,7 +120,8 @@ Newline5: NEWLINE
 Leading_asterisk3: LEADING_ASTERISK -> type(LEADING_ASTERISK);
 XmlTagOpen1: '<' -> type(OPEN), pushMode(xmlTagDefinition);
 STRING: '"' .*? '"' {referenceCatched = false;} -> mode(DEFAULT_MODE);
-PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {referenceCatched = true;};
+PACKAGE: [a-z_$] ([a-z0-9_$] | '.')+ [a-z0-9_$] {_input.LA(1) == '.' || _input.LA(1) == '}'
+                     || Character.isWhitespace(_input.LA(1))}? {referenceCatched = true;};
```

Comment:
@romani my input lacks _quotes_. My example isn't parsed without error even from javadoc tool but site is generated nonetheless. Please have a look at [PR 4550 comment](https://github.com/checkstyle/checkstyle/pull/4550#issuecomment-312521474).

Without context how can string be separated from package ? I `@ see https://...` will result in treatment of `https` as package without the predicate I added, since there are no double quotes to tell parser that it is rather a string. I feel this is amongst the most efficient way to distinguish between the two.